lib/turnout/redis_client.rb

```diff
@@ -0,0 +1,28 @@
+begin
+  require 'redis'
+rescue LoadError
+end
+
+module Turnout
+  class RedisClient
+    @@client = nil
+
+    def self.client
+      return @@client if @@client
+      if Gem.loaded_specs.has_key?('redis') && Turnout.config.redis_url != nil
+        @@client = Redis.new(url: Turnout.config.redis_url)
+      end
+    end
+
+    def self.maintenance?(settings)
+      return false unless self.client
+      message = self.client.get("turnout:maintenance")
+      if (message.is_a? String) && (message != "default")
+        settings.change_reason(message)
```

Comment:
Continuing on from before, it would be nice to support storing more than just the reason in Redis.